### ISSUE TITLE
Bug: Parser rejects computed keys in enum members

### ISSUE THREAD

**Issue by overlookmotel** [typescript-eslint/typescript-eslint#11795]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

parser

### Playground Link

https://typescript-eslint.io/play/#ts=5.9.2&fileType=.tsx&code=KYOwrgtgBAolDeAoKUDaByAHugulAvFAIwA0iAvokA&eslintrc=N4KABGBEBOCuA2BTAzpAXGYBfEWg&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA&tokens=false

### Repro Code

```TypeScript
enum E {
  ['x'] = 1,
}
```

### ESLint Config

```javascript

```

### tsconfig

```jsonc

```

### Expected Result

Parsing should succeed.

TypeScript appears to consider this valid syntax. [TS Playground](https://www.typescriptlang.org/play/?target=99#code/KYOwrgtgBAolDeAoKUDaByAHugulAvFAIwA0iAvokA)

### Actual Result

```
Parsing error: Computed property names are not allowed in enums. 2:2 - 2:3
```

### Additional Info

Apologies if there's some reason this isn't supported that I'm unaware of. I couldn't see any reference to this in issues (sorry if I've just not looked hard enough!).

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/parser`            | `8.48.0` |

**Comment by fisker** [comment]
Intentionally https://github.com/typescript-eslint/typescript-eslint/pull/11232#discussion_r2135160712

**Comment by overlookmotel** [comment]
Thanks very much @fisker. All is clear now!

FYI Oxc does support this syntax. 🤡

**Comment by fisker** [comment]
> FYI Oxc does support this syntax. 

I know, go kill it!

I didn't ask oxc to reject, since current TS version still allow.


https://github.com/prettier/prettier/blob/864bb7c042d9935344c155ad9292de3213e0085b/tests/format/misc/errors/typescript/enum/format.test.js#L17
